cli/flox-rust-sdk/src/providers/buildenv.rs

```diff
@@ -198,7 +203,21 @@ where
             }
         }
 
-        check_store_paths(all_paths)
+        let checked_store_paths = loop {
+            let checked_store_paths = check_store_paths(&all_paths)?;
+            match create_gc_root_in(
+                &checked_store_paths.valid,
+                self.new_gc_root_path("pre-checked-paths"),
```

Comment:
blocking question: did you check if this has a significant impact on performance?